src/views/Zap/__tests__/ZapStakeAction.unit.test.tsx

```diff
@@ -2,104 +2,44 @@ import { fireEvent } from "@testing-library/dom";
-      balanceUSD: 10000.0,
-    },
-    eth: {
-      address: "0x0000000000000000000000000000000000000000",
-      decimals: 18,
-      hide: false,
-      tokenImageUrl:
-        "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0x0000000000000000000000000000000000000000.png",
-      symbol: "ETH",
-      network: "ethereum",
-      price: 3397.72,
-      balance: 1,
-      balanceRaw: "1",
-      balanceUSD: 3397.72,
-    },
-  },
-};
-
-beforeEach(() => {
-  jest.useFakeTimers();
-});
-
 afterEach(() => {
-  jest.clearAllMocks();
+  jest.resetAllMocks();
   jest.restoreAllMocks();
 });
 
 describe("<ZapStakeAction/> ", () => {
   beforeEach(() => {
     const data = jest.spyOn(useWeb3Context, "useWeb3Context");
     data.mockReturnValue(mockWeb3Context);
-
-    jest.spyOn(useZapTokenBalances, "useZapTokenBalances").mockReturnValue(mockZapTokenBalances(tokenBalances));
-
-    jest.spyOn(usePrices, "useOhmPrice").mockReturnValue(mockOhmPrice(32));
-
-    jest.spyOn(usePrices, "useGohmPrice").mockReturnValue(mockGohmPrice(3400));
-
-    jest.spyOn(useBalance, "useSohmBalance").mockReturnValue(
-      mockSohmBalance({
-        [NetworkId.MAINNET]: new DecimalBigNumber("10"),
-        [NetworkId.TESTNET_RINKEBY]: new DecimalBigNumber("0"),
-      }),
-    );
-
-    jest.spyOn(useBalance, "useGohmBalance").mockReturnValue(
-      mockGohmBalance({
-        [NetworkId.MAINNET]: new DecimalBigNumber("10"),
-        [NetworkId.TESTNET_RINKEBY]: new DecimalBigNumber("0"),
-        [NetworkId.ARBITRUM]: new DecimalBigNumber("0"),
-        [NetworkId.ARBITRUM_TESTNET]: new DecimalBigNumber("0"),
-        [NetworkId.AVALANCHE]: new DecimalBigNumber("0"),
-        [NetworkId.AVALANCHE_TESTNET]: new DecimalBigNumber("0"),
-        [NetworkId.POLYGON]: new DecimalBigNumber("0"),
-        [NetworkId.FANTOM]: new DecimalBigNumber("0"),
-        [NetworkId.OPTIMISM]: new DecimalBigNumber("0"),
-      }),
-    );
-
-    jest.spyOn(useContractAllowance, "useContractAllowance").mockReturnValue(mockContractAllowance(BigNumber.from(10)));
+    global.fetch = jest.fn().mockResolvedValue({ ok: true, json: jest.fn().mockReturnValue(zapAPIResponse) });
+    //@ts-expect-error
+    Balances.useSohmBalance = jest.fn().mockReturnValue({ 1: { data: new DecimalBigNumber("10") } });
+    //@ts-expect-error
+    Balances.useGohmBalance = jest.fn().mockReturnValue({ 1: { data: new DecimalBigNumber("10") } });
+    //@ts-expect-error
+    Prices.useGohmPrice = jest.fn().mockReturnValue({ data: "3400.00" });
+    //@ts-expect-error
+    Prices.useOhmPrice = jest.fn().mockReturnValue({ data: "32.00" });
+    //@ts-expect-error
+    ContractAllowance.useContractAllowance = jest.fn().mockReturnValue({ data: BigNumber.from(0) });
   });
 
   it("gOHM should autopopulate with correct value based on ETH input", async () => {
+    //@ts-expect-error
+    ContractAllowance.useContractAllowance = jest.fn().mockReturnValue({ data: BigNumber.from(10) });
```

Comment:
Shouldn't the default be that there is a token allowance?

Reply:
you're right on this one. I had previously tested another scenario with needing to approve DAI, but no need for an approval for ETH